docs/features/techdocs/addons.md

```diff
@@ -0,0 +1,214 @@
+discover available Addons, we've compiled a list of them here:
+
+| Addon                                                                                                      | Package/Plugin                                     | Description                                                                                                                                                                                                                                  |
+| ---------------------------------------------------------------------------------------------------------- | -------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| [`<ReportIssue />`](https://backstage.io/docs/reference/plugin-techdocs-module-addons-contrib.reportissue) | `@backstage/plugin-techdocs-module-addons-contrib` | Allows TechDocs users to select a portion of text on a TechDocs page and open an issue against the repository that contains the documentation, populating the issue description with the selected text according to a configurable template. |
+
+Got an Addon to contribute? Feel free to add a row above!
+
+## Creating an Addon
+
+The simplest Addons are plain old react components that get rendered in
+specific locations within a TechDocs site. To package such a react component as
+an Addon, follow these steps:
+
+1. Write the component in your plugin like any other component
+2. Create, provide, and export the component from your plugin
+
+```ts
+// plugins/your-plugin/src/plugin.ts
+
+import {
+  createTechDocsAddonExtension,
+  TechDocsAddonLocations,
+} from '@backstage/plugin-techdocs-react/alpha';
+import { CatGifComponent, CatGifComponentProps } from './addons';
+
+// ...
+
+// You must "provide" your Addon, just like any extension, via your plugin.
+export const CatGif = yourPlugin.provide(
+  // This function "creates" the Addon given a component and location. If your
+  // component can be configured via props, pass the prop type here too.
+  createTechDocsAddonExtension<CatGifComponentProps>({
+    name: 'CatGif',
+    location: TechDocsAddonLocations.Header,
+    component: CatGifComponent,
+  }),
+);
+```
+
+### Addons in the Content location
+
+Beyond the "render a component in a region" use-case, it's also possible for
+Addons to access and manipulate a TechDocs site's DOM; this could be used to,
+for example, load and instantiate client-side diagramming libraries, replace
+elements with dynamically loaded content, etc.
+
+This type of Addon is still expressed as a react component, but instead of
+returning a react element to be rendered, it updates the DOM via side-effects
+(e.g. with `useEffect`). Access to the DOM is made available via utility hooks
+provided by the Addon framework.
```

Comment:
This is awesome! So we can actually get rid of the virtual `Component` location and simply use `Content` only 👍

Reply:
I do like `Component` as a future improvement -- it would simplify a lot of use-cases we've built up internally.

But I'll create a follow-up issue to track it.

Reply:
https://github.com/backstage/backstage/issues/11109